### ISSUE TITLE
Drag-Drop issues

### ISSUE THREAD

**Issue by mickev36** [Clariity/react-chessboard#32]
Hello,

I installed a bare bones project following the README.
Unfortunately, there seems to be an issue with the drag-and-drop : while dragging a piece, it disappears completely, and reappears only when releasing it on a square.

It sometimes kinda works, as the piece is semi-transparent, but the behaviour seems random (moving the same pieces makes it disapeear while dragging again).

Is there any configuration I'm missing ?

EDIT : I've cloned this repo and ran the examples, no issue. So it's probably something related to my project. I've ran a simple create-react-app, but with a typescript template. Could it be linked to typescript ?

EDIT2 : Tried with a vanilla JS create-react-app, same issue. Probably not related to typescript.

**Comment by Clariity** [comment]
Perhaps it's another factor?

What node version are you using?
What OS are you using?
What browser are you using?

What version of the package are you downloading? Are you using drag and drop elsewhere that may be interfering?

**Comment by mickev36** [comment]
I'm using : 
Node v16.14.0 / npm 8.10.0
I'm on a standard Ubuntu
I've tried firefox 101.0.1 and Chrome, same issue on both.

I've tried to use the "package" folder provided in the examples directly in my project (with file://), and have the same issue.
The project is just a bare bones create-react-app, where I only include this lib, so i don't think anything is interfering.

I'm really puzzled, since the example project is working on my machine, and i don't see any difference with a fresh create-react-app.

The only difference I noticed is I use React 18, whereas the example project uses React 17. Maybe something related to react-dnd ?

Please let me know if any more information could be useful

**Comment by mickev36** [comment]
I've tracked down the issue : I've downgraded my bare-bones create-react-app to use React v17. No issue at all. It seems like react-chessboard is not compatible with react 18, at least not the drag & drop.

**Comment by Clariity** [comment]
Yeah I thought it might be something like that.

I'll see if I can replicate it on my Windows machine once I am home from work, and then notify React DnD, although it hasn't been maintained in a while and doesn't look like it will be :( 

Thank you for raising this for me. Hopefully there will be a solution in the future

**Comment by mickev36** [comment]
Thank you for the quick responses and your work !
You'll be able to replicate it easily by using the standard create-react-app and importing and adding the lib to the project. Unfortunate that React DnD is not maintained anymoyre :-(

Good luck for the fixes :-)

**Comment by mickev36** [comment]
Understood, that's unfortunate. I moved to using chessground right now which fits my needs. I'll keep an eye on this repo  in the future though. Good luck, hope you'll find a fix !

**Comment by Clariity** [comment]
@mickev36 

I've remembered that we have 2 custom props that may solve your issues for you:

customDndBackend
customDndBackendOptions

this allows you to pass in your own DnD backend to handle drag and drop options in the event you can find one that works for your use case

**Comment by mickev36** [comment]
Thanks for the tip. I've implemented a solution with chessgrounds which fits my needs right now, but I might come back to react-chessboard in the future depending on the direction my project will take. If that's the case, I will try that solution and will keep you updated

**Comment by Clariity** [comment]
@mickev36 Fixed in 1.2.6

https://github.com/Clariity/react-chessboard/pull/39

**Comment by mickev36** [comment]
Awesome, thanks, i'll take a look !

**Comment by Frosty92** [comment]
Hi @Clariity,

I've installed version 1.2.6 of this module but the drag and drop issue seems to persist and is identical to @mickev36.  Here are my dependencies:   {
    "@react-chess/chessground": "^1.3.0",
    "@types/chess.js": "^0.13.2",
    "chess.js": "^0.13.4",
    "chessboardjsx": "^2.4.7",
    "next": "12.2.3",
    "react": "18.2.0",
    "react-chessboard": "^1.2.6",
    "react-dom": "18.2.0"
  }


Any suggestions ?

**Comment by Clariity** [comment]
Reopening as I'm able to replicate the bug, and boy is it a weird bug

So I can get drag and drop to work fine with the following dependencies

WORKING:
```
"dependencies": {
    "chess.js": "^0.12.0",
    "react": "^18.2.0",
    "react-chessboard": "^1.2.7",
    "react-dom": "^18.2.0",
    "react-scripts": "^5.0.1"
  },
  "devDependencies": {
    "eslint": "^7.32.0",
    "eslint-config-standard": "^16.0.3",
    "eslint-plugin-import": "^2.24.0",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-promise": "^5.1.0",
    "eslint-plugin-react": "^7.24.0"
  },
```

but with a fresh create-react-app with these dependencies, there is an issue

NOT WORKING:
```
"dependencies": {
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^13.3.0",
    "@testing-library/user-event": "^13.5.0",
    "react": "^18.2.0",
    "react-chessboard": "^1.2.7",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "web-vitals": "^2.1.4"
  },
```

So it's time for trial and error to work out what the culprit is, any help would be appreciated because this is bizaare

**Comment by Clariity** [comment]
I'll need a few days as I'm busy for a few evenings but then I can look into upgrading my react-dnd implementation as it's quite out of date now

**Comment by Clariity** [comment]
if anyone else wants to give it a go in the mean time then please do, you just need to upgrade `react-dnd`, `react-dnd-html5-backend` and `react-dnd-touch-backend`, and then fix the DragLayer implementation

**Comment by Clariity** [comment]
https://react-dnd.github.io/react-dnd/docs/tutorial#add-a-drag-preview-image

**Comment by Clariity** [comment]
@Frosty92 @mickev36 

Actually fixed it now in 1.2.8 :))

Took a while but I got there

**Comment by Frosty92** [comment]
Hi @Clariity , I tested it and it's working - thank you so much!
